glue/flagging_script_glue/flagging.py

```diff
@@ -92,36 +93,29 @@ def outlier_taxonomy(df: pd.DataFrame, permut: tuple, groups: tuple, condos: boo
     return df
 
 
-def iso_forest(
-    df: pd.DataFrame,
-    groups: tuple,
-    columns: list,
-    n_estimators: int = 1000,
-    max_samples: int or float = 0.2,
-) -> pd.DataFrame:
+def iso_forest(df, groups, columns, n_estimators=1000, max_samples=0.2):
```

Comment:
I haven't fully followed the changes needed for this repo, so can you just briefly add context on the purpose of this change?

Reply:
@dfsnow Posting an example of the error below. It comes from this line:
` df["sv_anomaly"] = isof.fit_predict(feed)`, For some reason, the current group iteration of iso forest can't handle a non-numeric value, such as `"Lake View 1"` (new geography name). The change here temporarily encodes it into a numeric type using an encoder, and flips it back at the end of the function. 

``` ValueError                                Traceback (most recent call last)
<ipython-input-95-9c01be93fa34> in ?()
----> 1 df_res_multi_fam_flagged = flg_model.go(
      2     df=df_res_multi_fam_to_flag,
      3     groups=tuple(inputs["stat_groups"]["multi_family"]),
      4     iso_forest_cols=inputs["iso_forest"],

~/repos/ccao_sales_val/glue/flagging_script_glue/flagging.py in ?(df, groups, iso_forest_cols, dev_bounds, condos)
     46     df = create_stats(df, groups, condos=condos)  # 'year', 'township_code', 'class'
     47     print("create_stats() done")
     48     df = string_processing(df)
     49     print("string_processing() done")
---> 50     df = iso_forest(df, groups, iso_forest_cols)
     51     print("iso_forest() done")
     52     df = outlier_taxonomy(df, dev_bounds, groups, condos=condos)
     53     print("outlier_taxonomy() done\nfinished")

~/repos/ccao_sales_val/glue/flagging_script_glue/flagging.py in ?(df, groups, columns, n_estimators, max_samples)
    127         max_samples=max_samples,
    128         bootstrap=True,
    129         random_state=42,
    130     )
--> 131     df["sv_anomaly"] = isof.fit_predict(feed)
    132 
    133     df["sv_anomaly"] = np.select(
    134         [(df["sv_anomaly"] == -1), (df["sv_anomaly"] == 1)],

~/repos/ccao_sales_val/venv/lib/python3.10/site-packages/sklearn/base.py in ?(self, X, y)
   1031         y : ndarray of shape (n_samples,)
   1032             1 for inliers, -1 for outliers.
   1033         """
   1034         # override for transductive outlier detectors like LocalOulierFactor
-> 1035         return self.fit(X).predict(X)

~/repos/ccao_sales_val/venv/lib/python3.10/site-packages/sklearn/base.py in ?(estimator, *args, **kwargs)
   1147                 skip_parameter_validation=(
   1148                     prefer_skip_nested_validation or global_skip_validation
   1149                 )
   1150             ):
-> 1151                 return fit_method(estimator, *args, **kwargs)

~/repos/ccao_sales_val/venv/lib/python3.10/site-packages/sklearn/ensemble/_iforest.py in ?(self, X, y, sample_weight)
    286         -------
    287         self : object
    288             Fitted estimator.
    289         """
--> 290         X = self._validate_data(X, accept_sparse=["csc"], dtype=tree_dtype)
    291         if issparse(X):
    292             # Pre-sort indices to avoid that each individual tree of the
    293             # ensemble sorts the indices.

~/repos/ccao_sales_val/venv/lib/python3.10/site-packages/sklearn/base.py in ?(self, X, y, reset, validate_separately, cast_to_ndarray, **check_params)
    600                 out = y
    601             else:
    602                 out = X, y
    603         elif not no_val_X and no_val_y:
--> 604             out = check_array(X, input_name="X", **check_params)
    605         elif no_val_X and not no_val_y:
    606             out = _check_y(y, **check_params)
    607         else:

~/repos/ccao_sales_val/venv/lib/python3.10/site-packages/sklearn/utils/validation.py in ?(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, estimator, input_name)
    914                         )
    915                     array = xp.astype(array, dtype, copy=False)
    916                 else:
    917                     array = _asarray_with_order(array, order=order, dtype=dtype, xp=xp)
--> 918             except ComplexWarning as complex_warning:
    919                 raise ValueError(
    920                     "Complex data not supported\n{}\n".format(array)
    921                 ) from complex_warning

~/repos/ccao_sales_val/venv/lib/python3.10/site-packages/sklearn/utils/_array_api.py in ?(array, dtype, order, copy, xp)
    376         # Use NumPy API to support order
    377         if copy is True:
    378             array = numpy.array(array, order=order, dtype=dtype)
    379         else:
--> 380             array = numpy.asarray(array, order=order, dtype=dtype)
    381 
    382         # At this point array is a NumPy ndarray. We convert it to an array
    383         # container that is consistent with the input's namespace.

~/repos/ccao_sales_val/venv/lib/python3.10/site-packages/pandas/core/generic.py in ?(self, dtype)
   1992     def __array__(self, dtype: NpDtype | None = None) -> np.ndarray:
-> 1993         return np.asarray(self._values, dtype=dtype)

ValueError: could not convert string to float: 'Lake View 5
```